uk_election_timetables/election.py

```diff
@@ -109,6 +108,7 @@ def get_date_for_event_type(self, event):
         for e in self.timetable:
             if e["event"] == event.name:
                 return e["date"]
+        raise KeyError("event not found")
```

Comment:
Fundamentally ruff is right that this code is a bit crap. I decided to `raise` here instead of just adding an explicit `return None`.
This isn't strictly a `KeyError` because timetable isn't a dict, but it feels like a `KeyError` in spirit. Open to other views on this. Is there a more appropriate error for "the thing you were looking for isn't in this data structure (that is not a dictionary)"

Reply:
Tricky one, but I don't think I care enough to argue about changing it! This is clearly an improvement. I guess the slightly more generic `ValueError` might be right, but....meh, this is fine